tests/test_edgeql_casts.py

```diff
@@ -2631,3 +2631,24 @@ async def test_edgeql_casts_all_null(self):
                 msg=f'{cast.from_type.name} to {cast.to_type.name}',
                 json_only=json_only,
             )
+
+    async def test_edgeql_casts_uuid_to_object(self):
+        persons = await self.con.query('select Person { id }')
+
+        res = await self.con.query('select <Person><uuid>$0', persons[0].id)
+        self.assertEqual(1, len(res))
+
+        async with self.assertRaisesRegexTx(
```

Comment:
Add a test where we apply it to a set of ids where *one* of the ids is missing and the rest are there, to make sure that that raises an error.

Reply:
Ha. It turns out that this does not raise an error. Did I do something wrong with contexts?

Reply:
Ok, I think we do need to make sure that the set appears in the scope tree, like this:
https://github.com/edgedb/edgedb/blob/master/edb/edgeql/compiler/casts.py#L480-L483

When you test this out, make sure to take a look at the scope tree with and without doing this, to try to see what's going on?

Reply:
(Interestingly, I bet that if you had `with` bound the set you were casting, and then cast that *name*, it would have worked. This is because compiling that *name* would have put the path in the scope tree)

Reply:
Hmmm, after I added UUID into the error message this now passes. Is this the consequence of the strange scope tree rules?

Reply:
Ha, yeah.
In the original take, the reference to the anchor appeared only in a subquery. (And like, triply so: inside of SET OF argument, inside a select statement, inside a filter clause thereof.)
But once the error message was added, it was referred to in the scope *outside* of the SET OF argument to `assert_exists`, so that is where it gets bound...